src/Symfony/Component/Validator/Constraints/CssColorValidator.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Validator\Constraints;
+
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\ConstraintValidator;
+use Symfony\Component\Validator\Exception\UnexpectedTypeException;
+use Symfony\Component\Validator\Exception\UnexpectedValueException;
+
+/**
+ * @author Mathieu Santostefano <msantostefano@protonmail.com>
+ */
+class CssColorValidator extends ConstraintValidator
+{
+    private const PATTERN_HTML5 = '/^#[0-9a-f]{6}$/i';
+    private const PATTERN_LONG = '/^#[0-9a-f]{8}$/i';
+    private const PATTERN_SHORT = '/^#[0-9a-f]{3,4}$/i';
```

Comment:
Why alpha is optional in "short pattern" but is either required or forbiden in "long pattern"?